10_rest/runnerdave/pkg/puppy/rest.go

```diff
@@ -0,0 +1,84 @@
+package puppy
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	chi "github.com/go-chi/chi"
+)
+
+// RestServer store for puppies
+type RestServer struct {
+	Db Storer
+}
+
+// SetupRoutes setups the routes for the server
+func (rs *RestServer) SetupRoutes(db Storer) *chi.Mux {
+	r := chi.NewRouter()
+	rs.Db = db
+
+	r.Route("/api/puppy", func(r chi.Router) {
+		r.Post("/", rs.createPuppy)
+		r.Route("/{id}", func(r chi.Router) {
+			r.Put("/", rs.updatePuppy)
+			r.Delete("/", rs.deletePuppy)
+			r.Get("/", rs.getPuppy)
+		})
+	})
+
+	return r
+}
+
+func (rs *RestServer) getPuppy(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err == nil {
+		p, rerr := rs.Db.ReadPuppy(int16(id))
+		if rerr != nil {
+			w.WriteHeader(http.StatusNotFound)
+		}
+		w.Header().Set("Content-Type", "application/json")
+		bytes, _ := json.Marshal(p)
+		i, _ := w.Write(bytes)
+		w.Header().Set("Content-Length", string(i))
+	}
+}
+
+func (rs *RestServer) createPuppy(w http.ResponseWriter, r *http.Request) {
+	post := Puppy{}
```

Comment:
nit: post is not a good name for this variable. You use `p` in `updatePuppy` - that's probably fine.